### ISSUE TITLE
Make mirage-crypto-ec (test) independent of mirage-crypto-pk

### ISSUE THREAD

**Issue by hannesm** [mirage/mirage-crypto#199]
As a quick fix for testing DKML and running the tests...

shouldn't be too hard, only the wycheproof stuff depends on:
```OCaml
let parse_sig cs =
  let asn = Asn.S.(sequence2 (required integer) (required integer)) in
  match Asn.(decode (codec der asn) cs) with
  | Error _ -> Error "ASN1 parse error"
  | Ok ((r, s), rest) ->
    if Cstruct.length rest <> 0 then Error "ASN1 leftover"
    else if Z.sign r < 0 || Z.sign s < 0 then
      Error "r and s must be >= 0"
    else
      Ok Mirage_crypto_pk.Z_extra.(to_cstruct_be r, to_cstruct_be s)
```

**Comment by hannesm** [comment]
hmm, and a bit more asn...

**Comment by hannesm** [comment]
fixed by #200
